backend/inspirehep/search/api.py

```diff
@@ -324,6 +324,27 @@ def get_author_papers(author, source=None, size=10000):
             results = results.params(_source=source)
         return results
 
+    def disambiguate_author(self, author):
+        emails = author.get("emails", [])
+        ids = author.get("ids", [])
+        emails_queries = [
+            (Q("match", email_addresses__value=email)) for email in emails
+        ]
+        id_queries = [
+            Q(
+                "bool",
+                must=[
+                    Q("match", ids__value=id_value.get("value")),
+                    Q("match", ids__schema=id_value.get("schema")),
+                ],
```

Comment:
this is incorrect, because `value` and `schema` don't necessarily go together as the field is not `nested`. That's why we use a validator for matching records: https://github.com/inspirehep/inspire-matcher/blob/0ca79e7d758dea2f6ee273dcbf7b9fa15f3c4dbb/inspire_matcher/config.py#L52 (one extra reason to use `inspire-matcher`).